action.yml

```diff
@@ -0,0 +1,46 @@
+name: mergeabot
+description: Auto-merge Dependabot PRs, only after quarantine days
+
+inputs:
+  quarantine-days:
+    description: |
+      How long PRs must have gone since their last update to qualify for
+      auto-merge. Default is 5, Set to -1 to disable
+    required: true
+    default: 5
```

Comment:
How do `default` and `required` play with each other? Does it make sense to have a default if I'm required to set this?

Reply:
> How do default and required play with each other? 

I think the best way to look at it is each in isolation:

- `required: true`: When you ask for it, you will either get a value or an error, you will never get an implicit `""` or `null`
- `default: x`: When they don't specify it, behave as if the user specified `x`

Conceptually, I want both of those behaviors individually, so this is how I tend to do it (and have seen it done).

>  Does it make sense to have a default if **I'm required to set** this?

As described above, that's not really how `required` works. _Someone_ is required to set it, including via `default`.

Reply:
Yeah, it's a bit of a cluster.

In the JS libs, when you access an input you can also pass,

```js
core.getInput("foo", { required: true })
```

And I have _no idea_ how that plays with the stuff from `action.yml`.